### ISSUE TITLE
Pong game breaks when using Airpods Pros

### ISSUE THREAD

**Issue by JuanLassoVelasco** [amethyst/amethyst#2698]
## Description

Changes made in the "Adding Audio" section of the pong tutorial seem to cause the game to break altogether when I connect my airpods to my computer. However, when I disconnect my airpods and use the built in speakers on my laptop, the game runs just fine. I haven't yet tested this with other bluetooth headphones so I can't say if it's a problem with all bluetooth audio devices.

## Reproduction Steps

1. Build up the pong game all the way up to "Winning Rounds and Keeping Score"
2. Verify the game works with or without airpods on
3. Then make the changes in the "Adding Audio" section
4. Try to run the game again with airpods on

## What You Expected to Happen
The game to run as normal with the sound added

## What Actually Happened
The game compiles but then instantly "crashes" (for lack of a better term) and throws this error

## Screenshots and Logs

When the game crashes the following output is shown in the terminal

'''
warning: unused imports: `World`, `core::SystemDesc`
 --> src/systems/move_ball.rs:4:5
  |
4 |     core::SystemDesc,
  |     ^^^^^^^^^^^^^^^^
5 |     derive::SystemDesc,
6 |     ecs::{Join, Read, ReadStorage, System, SystemData, World, WriteStorage},
  |                                                        ^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default

warning: unused imports: `SystemData`, `SystemDesc`, `World`, `derive::SystemDesc`
 --> src/systems/bounce.rs:2:23
  |
2 |     core::{Transform, SystemDesc},
  |                       ^^^^^^^^^^
3 |     derive::SystemDesc,
  |     ^^^^^^^^^^^^^^^^^^
4 |     ecs::{Join, ReadStorage, System, SystemData, World, WriteStorage},
  |                                      ^^^^^^^^^^  ^^^^^

warning: 2 warnings emitted

    Finished dev [unoptimized + debuginfo] target(s) in 0.91s
     Running `target/debug/pong`
[INFO][amethyst::app] Initializing Amethyst...
[INFO][amethyst::app] Version: 0.15.3
[INFO][amethyst::app] Platform: x86_64-apple-darwin
[INFO][amethyst::app] Amethyst git commit: UNKNOWN
[INFO][amethyst::app] Rustc version: 1.53.0 Stable
[INFO][amethyst::app] Rustc git commit: 53cb7b09b00cbea8754ffb78e7e3cb521cb8af4b
thread 'main' panicked at 'No device name: BackendSpecific { err: BackendSpecificError { description: "core foundation unexpectedly returned null string" } }', /Users/juanlasso/.cargo/registry/src/github.com-1ecc6299db9ec823/rodio-0.11.0/src/engine.rs:113:46
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
'''

## Your Environment

Operating System: macOS Big Sur
Operating System Version: 11.4
Version of Amethyst: 0.15

**Comment by JuanLassoVelasco** [comment]
After testing with other bluetooth headphones that I have, I can confirm that the game does, in fact, break when you use other  bluetooth headphones that aren't airpods.

**Comment by JuanLassoVelasco** [comment]
I also tried running the game first, then connecting my headphones while the game was running. This doesn't break the game. Instead, my headphones refuse to connect while the game is running. Just an interesting note I thought I'd mention.

**Comment by RossMurr4y** [comment]
Confirmed this is occurring for me also. Disconnected my AirPods pros audio worked correctly through the MacBook speakers.

```toml
[dependencies.amethyst]
version = "0.15"
features = [ "metal" ]
```

```terminal
thread '<unnamed>' panicked at 'No device name: BackendSpecific { err: BackendSpecificError { description: "core foundation unexpectedly returned null string" } }', /Users/rossmurray/.cargo/registry/src/github.com-1ecc6299db9ec823/rodio-0.11.0/src/engine.rs:113:46
```

**Comment by linco95** [comment]
This seems to be fixed by https://github.com/RustAudio/rodio/pull/280 which is released to rodio 0.12.0.
Updating rodio in amethyst_audio could be a potential fix for this.

I get the same problem on macOS Big Sur. It works with my SteelSeries headphones, but not with my external speakers.
